### PR TITLE
Resolve next major comments

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

**Comment by VincentLanglet** [comment]
Seems like some deprecations was forgotten in https://github.com/sonata-project/twig-extensions/pull/90

Since FlashManager doesn't implements StatusClassRendererInterface, it cannot be passed to the StatusRuntime anymore by the CompilerPass. Also, support for string value by the `statusClass` method doesn't make sens anymore. Do you agree @wbloszyk ?

One issue is that now, we don't have any implementation of the StatusClassRendererInterface. Which means that the statusClassRuntime is kinda useless like this. Maybe instead of removing the method from the FlashManager, we should just move them to a StatusClassRenderer instead. WDYT @jordisala1991 ?

Edit: Looking at the implementation of the FlashManager
https://github.com/sonata-project/twig-extensions/blob/1.x/src/FlashMessage/FlashManager.php#L109-L111
Seems like almost all objects were not handled so far, so this work like before.

File: src/Bridge/Symfony/DependencyInjection/SonataTwigExtension.php [hunk 1]
```diff
@@ -45,22 +45,21 @@ public function load(array $configs, ContainerBuilder $container): void
...
         $mergedConfig = array_merge_recursive($config['flashmessage'], [
             'success' => ['types' => [
-                'success' => [],
-                'sonata_flash_success' => [],
-                'sonata_user_success' => [],
-                'fos_user_success' => [],
+                'success',
+                'sonata_flash_success',
+                'sonata_user_success',
+                'fos_user_success',
```
Comment (jordisala1991) [inline src/Bridge/Symfony/DependencyInjection/SonataTwigExtension.php]: should we remove this?

Comment (wbloszyk) [reply src/Bridge/Symfony/DependencyInjection/SonataTwigExtension.php]: `fos_user_success` should be removed.

File: src/Extension/StatusRuntime.php [hunk 2]
```diff
@@ -37,45 +35,10 @@ public function addStatusService(StatusClassRendererInterface $renderer): void
     }
 
     /**
-     * NEXT_MAJOR: Restrict $object to object.
-     *
-     * @param object|string $object     Object for StatusClassRenderer or string for FlashManager
-     * @param string|null   $statusType Object status type or Sonata flash message type
-     * @param string        $default    Default status class
+     * @param string|null $statusType Object status type or Sonata flash message type
```
Comment (wbloszyk) [inline src/Extension/StatusRuntime.php]:
```suggestion
     * @param string|null $statusType Object status type
```

**Comment by wbloszyk** [comment]
>

**Review by wbloszyk — COMMENTED** [review]
Little changes

**Comment by wbloszyk** [comment]
> Seems like some deprecations was forgotten in #90
> 
> Since FlashManager doesn't implements StatusClassRendererInterface, it cannot be passed to the StatusRuntime anymore by the CompilerPass. Also, support for string value by the `statusClass` method doesn't make sens anymore. Do you agree @wbloszyk ?

Correct.

> One issue is that now, we don't have any implementation of the StatusClassRendererInterface. Which means that the statusClassRuntime is kinda useless like this. Maybe instead of removing the method from the FlashManager, we should just move them to a StatusClassRenderer instead. WDYT @jordisala1991 ?

`StatusClassRendererInterface` should be used to write own `*StatusRenderer` class. [doc](https://docs.sonata-project.org/projects/twig-extensions/en/1.x/reference/status_helper/#define-a-service) and [example](https://github.com/sonata-project/ecommerce/blob/3.x/src/Component/Order/OrderStatusRenderer.php)

> Edit: Looking at the implementation of the FlashManager https://github.com/sonata-project/twig-extensions/blob/1.x/src/FlashMessage/FlashManager.php#L109-L111 Seems like almost all objects were not handled so far, so this work like before.

Correct. After add type hint this solution would not be work, so `FlashManager` and `StatusClassRenderer` had to be separated.

`StatusClassRenderer::getStatusClass` for display class in alert/badges(...) depends on object
`FlashManager` is service with flash functionality like `AbstractController::addFlash(string $type, $message)` - type is used as css class.

**Comment by jordisala1991** [comment]
So there is more work needed to close this one or is it ready?

**Comment by VincentLanglet** [comment]
> So there is more work needed to close this one or is it ready?

It's ready

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**